numpy/core/arrayprint.py

```diff
@@ -337,6 +354,29 @@ def _array2string(a, max_line_width, precision, suppress_small, separator=' ',
     return lst
 
 
+def _recursive_guard(fillvalue='...'):
```

Comment:
I think it would be good to add a comment here that this is to account for (for example) self-containing object arrays.